test/validation/api/system/system.c

```diff
@@ -208,17 +208,19 @@ static void system_test_cpu_cycles_long_period(void)
 	res = odp_cpu_cycles_resolution();
 	max = odp_cpu_cycles_max();
 
+	c3 = odp_cpu_cycles();
+
+	CU_ASSERT(c3 <= max);
 	/*
-	 * We can virtually never see a 64 bit cycle counter wrap around,
-	 * so let's not even try. Use small a number of periods to speed
-	 * up testing in this common case.
+	 * If the cycle counter is not even close to wrapping around during
+	 * the test, then speed up the test by not trying to see the wrap
+	 * around too hard.
 	 */
-	if (max == UINT64_MAX)
+	if (max - c3 > 10 * periods * 100 * (ODP_TIME_MSEC_IN_NS + periods))
```

Comment:
Looks a little cryptic. Maybe it doesn't matter as long as the result is big enough, but to correspond to the loop below, shouldn't this be `10 * periods * (100 * ODP_TIME_MSEC_IN_NS + periods / 2)`?

We are comparing here cpu cycles and nanoseconds, and assuming that there can't be more than 10 cycles in a nanosecond, right? If yes, then maybe this could be mentioned in the comment.

Reply:
Yes, the code is bad, although not dividing by two was intentional. Hopefully made it more clear in v3.